src/OrchardCore.Modules/OrchardCore.Queries/Liquid/QueriesLiquidTemplateEventHandler.cs

```diff
@@ -0,0 +1,24 @@
+using System.Threading.Tasks;
+using Fluid;
+using Fluid.Values;
+using OrchardCore.Liquid;
+
+namespace OrchardCore.Queries.Liquid
+{
+    public class QueriesLiquidTemplateEventHandler : ILiquidTemplateEventHandler
+    {
+        private readonly IQueryManager _queryManager;
+
+        public QueriesLiquidTemplateEventHandler(IQueryManager queryManager)
+        {
+            _queryManager = queryManager;
+        }
+
+        public Task RenderingAsync(TemplateContext context)
+        {
+            context.LocalScope.SetValue("Queries", new LiquidPropertyAccessor(async name => FluidValue.Create(await _queryManager.GetQueryAsync(name))));
```

Comment:
@sebastienros if you can get rid of `LiquidPropertyAccessor` then I'll remove it from here too.